[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98264
* __->__ #98088

Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

desertfire committed:
[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen

Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen"


Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen"


Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen"


Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

[ghstack-poisoned]

jgong5 commented on torch/_inductor/codegen/wrapper.py:
```diff
@@ -803,81 +784,53 @@ def generate(self):
-        _, _, kernel_path = get_code_path(source_code, ext, extra)
-        return kernel_path
-
-    def load_kernel(self, name: str = None, kernel: str = None, arg_types: List = None):
```
Also remove the call to `load_kernel` from `cpp.py`?

jgong5 reviewed (approved):
LGTM.

chunyuan-w commented on torch/_inductor/ir.py:
```diff
@@ -2022,9 +2022,6 @@ class NoneAsConstantBuffer(IRNode):
     def codegen_reference(self):
         return "None"
 
-    def cpp_wrapper_codegen_reference(self):
```
I'm working on supporting dynamic shapes for cpp_wrapper (https://github.com/pytorch/pytorch/pull/97965 in draft) and might need to add a `cpp_wrapper_codegen_reference` function to the `ShapeAsConstantBuffer` class as well.

desertfire replied:
Because we never generate mixed wrapper, I think using `V.graph.cpp_wrapper` will be sufficient to distinguish the behavior.

desertfire committed:
Update on "[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen"


Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen"


Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen"


Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -g

desertfire commented:
@pytorchbot merge

desertfire committed:
Update on "[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen"


Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Combine CppWrapperCodeGen and CppAotWrapperCodeGen"


Summary: Make CppAotWrapperCodeGen generate kernels and wrapper in one
file, which unifies the codegen for AOT and non-AOT mode. There will be
more refactoring for the AOT part.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge
